server/schedulers/hot_region.go

```diff
@@ -241,28 +249,28 @@ func (h *hotScheduler) updateStatsByPendingOpInfo(storeStatsMap map[uint64]float
 }
 
 func (h *hotScheduler) gcPendingOpInfos() {
-	for regionID, pendingOpInfos := range h.pendingOpInfos {
-		var n *list.Element
-		for e := pendingOpInfos.Front(); e != nil; e = n {
-			n = e.Next()
-			opInfo := e.Value.(*pendingInfluence)
-			if opInfo.isDone() {
-				if time.Now().After(opInfo.op.GetCreateTime().Add(statistics.StoreHeartBeatReportInterval * time.Second)) {
+	for regionID, pendings := range h.regionPendings {
+		empty := true
+		for ty, pending := range pendings {
+			if pending != nil && pending.isDone() {
```

Comment:
`pendings = {nil,p2}`, and  p2 is not  done, empty will be true.

Reply:
Fixed and added a test.